src/proto/diff.proto

```diff
@@ -66,6 +66,15 @@ message ChangedFile {
 		CS4 = 204;
 		/** @exclude */
 		CS5 = 205;
+		
+		/** The file represents a JavaScript source file that had a parse error */
+		SOURCE_JS_ERROR = 300;
+		/** The file represents a JavaScript source file that parsed without error as JLS2 */
+		SOURCE_JS_JLS2 = 301;
```

Comment:
The eclipse jsdt has two JLS fields under ast, JLS2 and JSL3. From the eclipse documentation JSL3 supports unto ECMA4 it is unclear what EMCA level JSL2 supports.
http://help.eclipse.org/kepler/index.jsptopic=%2Forg.eclipse.wst.jsdt.doc%2Freference%2Fapi%2Forg%2Feclipse%2Fwst%2Fjsdt%2Fcore%2Fdom%2FAST.html